### DISCUSSION TITLE
rounded-full on one side removes the round from the other side

### DISCUSSION THREAD

**Discussion by ahmedsayedabdelsalam** [tailwindlabs/tailwindcss#13080] [Help] [answered]
when i try to make rounded-t-full it removes the style of rounded-b-lg  [playground](https://play.tailwindcss.com/vQXmeVqie9?layout=horizontal)

<img src="https://github.com/tailwindlabs/tailwindcss/assets/29691074/f5eb86e1-56ef-4fc6-a7a7-78b36844db2d" width="500" />

**Comment by AnshSinghSonkhia** [comment]
You are correct @ahmedsayedabdelsalam, it's not working.

**Comment by RobinMalfait** [comment, marked as answer]
Hey!

The short answer is: `border-radius` is complex.

They behave relative to each other, the `rounded-full` is implemented as `9999px` which is very large compared to the `2rem` for example.
The reason we don't use `100%` or `50%` is because indeed it will not work as expected on non-square elements.

Here are a few solutions you can use: https://play.tailwindcss.com/rqciMqM2F0

**Reply by ahmedsayedabdelsalam** [reply]
> Hey!
> 
> 
> 
> The short answer is: `border-radius` is complex.
> 
> 
> 
> They behave relative to each other, the `rounded-full` is implemented as `9999px` which is very large compared to the `2rem` for example.
> 
> The reason we don't use `100%` or `50%` is because indeed it will not work as expected on non-square elements.
> 
> 
> 
> Here are a few solutions you can use: https://play.tailwindcss.com/rqciMqM2F0

Thanks 🙏
